src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3357,6 +3357,37 @@
     <Field Name="Keyword" Type="SyntaxToken" Override="true">
       <ContextualKind Name="RecordKeyword"/>
     </Field>
+    <Field Name="RecordModifier" Type="SyntaxToken" Optional="true">
+      <Kind Name="ClassKeyword"/>
+    </Field>
+    <Field Name="Identifier" Type="SyntaxToken" Override="true">
+      <Kind Name="IdentifierToken"/>
+    </Field>
+    <Field Name="TypeParameterList" Type="TypeParameterListSyntax" Optional="true" Override="true"/>
+    <Field Name="ParameterList" Type="ParameterListSyntax" Optional="true" />
+    <Field Name="BaseList" Type="BaseListSyntax" Optional="true" Override="true"/>
+    <Field Name="ConstraintClauses" Type="SyntaxList&lt;TypeParameterConstraintClauseSyntax&gt;" Override="true"/>
+    <Field Name="OpenBraceToken" Type="SyntaxToken" Override="true" Optional="true">
+      <Kind Name="OpenBraceToken"/>
+    </Field>
+    <Field Name="Members" Type="SyntaxList&lt;MemberDeclarationSyntax&gt;" Override="true"/>
+    <Field Name="CloseBraceToken" Type="SyntaxToken" Override="true" Optional="true">
+      <Kind Name="CloseBraceToken"/>
+    </Field>
+    <Field Name="SemicolonToken" Type="SyntaxToken" Optional="true" Override="true">
+      <Kind Name="SemicolonToken"/>
+    </Field>
+  </Node>
+  <Node Name="RecordStructDeclarationSyntax" Base="TypeDeclarationSyntax">
```

Comment:
How come we don't need RecordClassDecl? IMO this is a StructDecl with record as a modifier (similarly, a ClassDecl with record as a modifier), with that reasoning `record` itself stays a RecordDecl. I understand that record and record class are semantically identical, but I think it's irrelevant to the syntax nodes we're using..

Thinking about how partial fits into all this. do we have `partial record struct` or `record partial struct`? If we consider `record` a modifier as I'm suggesting, both would be valid (this involves parts of *relaxed modifier ordering* feature).

Reply:
> partial

The spec/grammar doesn't allow `record partial struct`. If we want to support that, then that would indeed push for `record` to be a modifier on a StructDecl.

Treating `record` as a modifier is worth some more though. I tend to see "record struct" as one word, but I know that we've regretted such decisions in the past... Will raise question with LDM.

Reply:
tbh, I couldn't tell which one would be disallowed with current design. for classes it's even worse as we already have `partial record,` so you can't just append `class` (for any reason) and have to reorder modifiers. Def worth considering.

Reply:
From chat with LDM, the preference is to keep the current grammar (https://github.com/dotnet/csharplang/blob/master/proposals/record-structs.md#record-structs), that is `record` and `struct` must be stuck together.

Reply:
Makes sense. Though that doesn't seem to need the syntax node to reflect that exact grammar (`partial class` is one word and it still use a modifier for partial).